bin/needs-a-genny

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env ruby
+
+genny_path = ->(slug) {
+  File.join 'exercises', slug, '.meta', 'generator'
+}
+
+no_genny_path = ->(slug) { !Dir.exist? genny_path[slug] }
+
+puts Dir.children('exercises')
```

Comment:
`exercises` means this script is only runnable from the top level of this repo, which is inconvenient to force someone to only be able to run it from one location.

thus, it may be preferable to use `File.join(__dir__, '..', 'exercises')` instead so that the script is runnable from anywhere.

Reply:
I didn't realize I could get `File.join` to do that. Question though: can the other tools in `bin/` be successfully run anywhere? If so then this should definitely conform to them.

Reply:
Are you saying that `File.join` will magically find `exercises/` by including `'..'`, or would it make it simply go to the parent like in shell?

Reply:
>can the other tools in bin/ be successfully run anywhere? If so then this should definitely conform to them.

`generate` will run from any folder, for example.  `local-status-check` will not, so it varies.

`setup-git-hooks` will not run from just anywhere, but it may make sense to be in the repository root while running this, at least minimally to be in the project folder somewhere.

Reply:
There is noting magical about File.join and `..`. `File.join(__dir__, '..', 'exercises')` will evaluate to someting like `"/path/to/repo/bin/../exercises"`, and `Dir.exercises("/path/to/repo/bin/../exercises")` will do as expected.

Reply:
There is nothing magical about File.join and `..`. `File.join(__dir__, '..', 'exercises')` will evaluate to something like `"/path/to/repo/bin/../exercises"`, and `Dir.children("/path/to/repo/bin/../exercises")` will do as expected.

Reply:
I tested it in irb, and it treats `..` just like bash would.  That is, if I do it from the repo root it makes a non-existent path string.  It seems like there is a way to do it, but I don't think the method you are suggesting works from anywhere (only if you are 1 directory deeper than the root).

Reply:
I needed `File.expand_path('../../exercises', __FILE__)`

Reply:
@petertseng FYI, `__dir__` seems to evaluate to `.` in my repl.

Reply:
> `__dir__` seems to evaluate to . in my repl.

that seems correct.